src/components/RestaurantPage/RestaurantPage.js

```diff
@@ -10,11 +10,20 @@ export class RestaurantPage extends React.Component {
   }
 
   componentDidMount() {
-    const { loadRestaurantInfo, match } = this.props;
+    const {
+      loadRestaurantInfo,
+      match,
+      loadLocationsVariants,
+      locationsVariants,
+    } = this.props;
 
     loadRestaurantInfo(match.params.id);
 
     window.scrollTo(0, 0);
+
+    if (!locationsVariants) {
+      loadLocationsVariants();
+    }
```

Comment:
as advanced task, you can create `ensureLocationsVariants` action based on `loadLocationsVariants` which will encapsulate logic of this `if`